### PR TITLE
Serve untrusted content with CSP enabled.

### PR DESCRIPTION
**Pull request by mpkorstanje**
The embeddings/ folder of cucumber-html-reports contains content that is created by the build process and should not be served withouth the proper precautions. To ensure this the CSP will be enabled for all files in cucumber-hml-reports unless their direct parent directory is white listed. The current white listed directories are:
 - /
 - /css
 - /fonts
 - /images
 - /js

### CODE REVIEW

**Comment by mpkorstanje** [comment]
Rough draft of the idea. 

safeDirectories should probebly be configurable.

File: src/main/java/net/masterthought/jenkins/SafeArchiveServingAction.java [hunk 1]
```diff
@@ -248,16 +264,38 @@ public void doDynamic(StaplerRequest req, StaplerResponse rsp) throws IOExceptio
...
-            rsp.serveFile(req, in, lastModified, -1, length, file.getName());
+    private HttpResponse serveFile(File file) throws IOException, ServletException {
+        if(safeDirectories.contains(file.getParentFile())){
+            // Reports in safe directories can be trusted and must be served
+            // without Content-Security-Policy to display reports properly
+            return new UnsafeDirectoryBrowserSupport(file);
+        } else {
+            // Other can (such as embeddings) can not be trusted,and must
+            // be served with Content-Security-Policy
+            return new DirectoryBrowserSupport(this, new FilePath(rootDir), title, iconName, false);
```
Comment (mpkorstanje) [inline src/main/java/net/masterthought/jenkins/SafeArchiveServingAction.java]: Using DirectoryBrowserSupport allows us to serve the file as if it was served from the workspace. This ensures the CSP is applied correctly without requiring us concern ourselves about the implementation details of doing so.

**Comment by damianszczepanik** [comment]
Thanks - do you recommend any other testing before releasing this code?

**Comment by mpkorstanje** [comment]
Didnt get around to properly testing regression yet. I am surpriced to see
it merged already.

Would recommend adding some non annoying js to the sample to show it works
in the iframe but not in the independent file.

Beyond that, no. It is a fairly simple change.

**Comment by damianszczepanik** [comment]
Would be nice, willing to update sample.json file? How about modifying https://github.com/damianszczepanik/cucumber-reporting/blob/master/src/test/resources/json/sample.json#L443 so it has one external image that should not be visible when loading over Jenkins and another one which does

**Comment by mpkorstanje** [comment]
No problem. I'll pick it up with the notice above the iframe.

**Pull request merged**